src/palace/manager/api/controller/base.py

```diff
@@ -108,21 +108,26 @@ def authenticate(self):
         data = self.manager.authentication_for_opds_document
         return Response(data, 401, headers)
 
-    def library_for_request(self, library_short_name):
-        """Look up the library the user is trying to access.
-
-        Since this is called on pretty much every request, it's also
-        an appropriate time to check whether the site configuration
-        has been changed and needs to be updated.
+    def library_for_request(
+        self, library_short_name: str | None
+    ) -> Library | ProblemDetail:
+        """Look up the library the user is trying to use.
+
+        :param library_short_name: Optional key for looking up the library.
+        :return: The Library for the provided key, if found; a default library,
+            if available; or the `LIBRARY_NOT_FOUND` ProblemDetail.
+
+        If the key is present, use it to look up the library. If it is not
+        found, return the `LIBRARY_NOT_FOUND` ProblemDetail. If the key is
+        not present, try to get a default library. If one is not found, return
+        the `LIBRARY_NOT_FOUND` ProblemDetail.
         """
-        self.manager.reload_settings_if_changed()
```

Comment:
This deletion is the key change. The additions are just adding some type hints and hopefully improving documentation.